### ISSUE TITLE
Lightberry + Philips Hue + Hyperion

### ISSUE THREAD

**Issue by apurbas** [hyperion-project/hyperion#556]
Hello All,
I'm using Lightberry on Rpi1 with Openelec (custom build available on Lightberry site). I'm struggling to control one more device ( 3 Philips Hue +1 bloom on old round bridge) using hyperion. I have created a config2 file for Hue, reusing the Lightberry config
http://www.jsoneditoronline.org/?id=e76e860e66731b65b3c3ce8cce8f6a9e

Proto/json servers are successfully created for config2 but hyperion couldn't control any of the hue lamps.
So far all I can debug is here - 
http://pastebin.com/0uSMrnbz

Can anyone help me in setting this up?

**Comment by ntim** [comment]
Mmh, color order has to be "rgb". Do you have the very latest version of the code from github?

**Comment by apurbas** [comment]
Hi @ntim - agreed, but the lightberry config surprisingly creates this color order. I think I have the latest one only. Could you suggest me a way to reverify this?

**Comment by apurbas** [comment]
Hello @ntim 
Changed the config to rgb without any effect :-1: 
Error seems -
hyperiond: /home/pi/fix_do_apa/hyperion/dependencies/build/jsoncpp/json_value.cpp:1136: Json::Value& Json::Value::resolveReference(const char*, bool): Assertion `type_ == nullValue || type_ == objectValue' failed.
Aborted

**Comment by ntim** [comment]
Ok, then follow the steps described https://github.com/tvdzwan/hyperion/issues/506 to produce a stack trace to see where the error is coming from.

**Comment by ntim** [comment]
You could also try my fork, it has a more precise output and more validity checks, will make a pull request as soon as I tested it extensively.

**Comment by apurbas** [comment]
Thanks @ntim. 
Some updates
I have changed the config file to following & named in config3.
http://www.jsoneditoronline.org/?id=e76e860e66731b65b3c3ce8cce8f6a9e
Its resolving & above errors are gone :+1: , but I find a new issue.
The setup runs lightberry strip or Philips Hue lamp flawless but when I try to start multiple hyperiond from command line, the Hue is static & throws following error.

http://pastebin.com/KrMtAPie
terminate called after throwing an instance of 'std::runtime_error'
  what():  VIDIOC_REQBUFS error 16, Device or resource busy
Aborted
Whats the issue here?

**Comment by ntim** [comment]
Mmh, I am not familiar with the lightberry but I guess you are only allowed to have one concurrent connection to the device as it is for most serial devices.

**Comment by apurbas** [comment]
Can I not use my video grabber (Easycap utv007) to create multiple hyperiond processes and thus  control multiple devices ( Lightberry strip & Philips Hue) as described [here](https://github.com/tvdzwan/hyperion/wiki/Controlling-multiple-devices)? 
Lightberry is mere strip (Apa 102). My system is on OpenElec 6.0.0 / Rpi1.

**Comment by tpmodding** [comment]
no you cannot use the videograbber multiple times...you have to use the forwarder... :/

**Comment by penfold42** [comment]
Good examples of forwarding here:
https://github.com/tvdzwan/hyperion/wiki/Network-and-distributed-features

**Comment by apurbas** [comment]
Thanks @ntim @tpmodding & @penfold42. 
I'm now using forwarder in following json ( removed grabber & added forwarder)

http://www.jsoneditoronline.org/?id=b1fd28c36609a2793fe0b0b6714df7da
This works fine i.e. config2 controls my hue [Light Id =4] &default config contols Lightberry/APA LED strip ONLY when I play a media on Kodi. 

http://pastebin.com/WPxW1hCe

But Somehow the 2nd config is not reading from my external video grabber (Easycap utv007) & accordingly not controlling other HDMI sources. Any idea what I'm doing wrong?

**Comment by tpmodding** [comment]
could you pls post all configs (apart) on pastebin

**Comment by apurbas** [comment]
@tpmodding Here is the original config for Lightberry setup.The external video grabber successfully renders light effects.
http://pastebin.com/6kU6W9d1

This is the config2 ( implementing forwarder etc) to control hue.
http://pastebin.com/WPxW1hCe

**Comment by tpmodding** [comment]
you kidding me? i need your config, not the output... you need 2 configs
first with your leds connected to your pi, usb grabber config, forwarder(where you SEND infos)
second with the device-type hue (where you RECEIVE infos)

ps: did you pay for lightberry? why they dont help you ;)

**Comment by apurbas** [comment]
oops my mistake
Here is default config http://pastebin.com/6kU6W9d1
Config 2 here http://pastebin.com/DyeCbwBi

**Comment by apurbas** [comment]
Implemented forwards. Could any one guide me why are these not working for external grabbers?
http://pastebin.com/1qtc6AWe
http://pastebin.com/b6wC0wj1
Works fine when I play a media on Kodi though..

**Comment by brindosch** [comment]
did you found a solution?

**Comment by apurbas** [comment]
Forward works but fails after some. Related issue (https://github.com/hyperion-project/hyperion/issues/578) & (https://github.com/hyperion-project/hyperion/issues/594)

**Comment by brindosch** [comment]
3 issues hehe

**Comment by apurbas** [comment]
Lets resolve this so that we have an updated hyperion forward wiki with all FAQs as well.

**Comment by brindosch** [comment]
//cleanup

**Comment by Portista1988** [comment]
Hello, 
I need some help. 
I habe a raspberry Pi3/video grabbeler/ and the philips Hue lamps.
I folled the steps to send the video to the PI so mine lights will sync but i can't get the lights link with the PI.
anybody any idea because i get stuck and have no idea anymore

**Comment by apurbas** [comment]
Can you post the config file please? Please run Hyperion through command
line. What error are you getting?


Max 2 Hue can be controlled and controlling hue requires adding a
'proto/json forward' code in the config files

**Comment by Portista1988** [comment]
Dear,

I first try to do it like paper said and look for the file.
But i get message that there is no file.
After i made one true what it says in the books with also mine info.
After it i try to start it and there is nothing.
See image downstairs.
Mine setup is with three light but if only with two works please let me know

I follewed every steps and both everything
I also can't get a screenshot or anything

I bought a Gembird Videograbber (that mine raspberry  finds when i play with lightberry on kodi)

![Command 1](https://user-images.githubusercontent.com/54452406/63646699-34788680-c717-11e9-8d2b-83978a3763c0.jpg)
![Command 2](https://user-images.githubusercontent.com/54452406/63646700-34788680-c717-11e9-98d3-980cddf351f9.jpg)



[hyperion.config.zip](https://github.com/hyperion-project/hyperion/files/3537910/hyperion.config.zip)

**Comment by apurbas** [comment]
Mate

Questions to you..

   1. What's your light set up. Are you using Old Hue bridge (round) or
   newer (Square)? Will Hyperion control 3 Hue bulbs only? Please go
through this
   <https://hyperion-project.org/wiki/PhilipsHue-Device>writeup
   2. Can you validate if Hyperion is installed in your Pi/ OpenElec? Refer
   to the troubleshooting section here <http://support.lightberry.eu/>.
   1. Start the Hyperion config from command line (Section 4.3). Do you see
      any error?
      2. Is your grabber up? Can Hyperion see that? If yes - try taking a
      screenshot using command
      3. Try without external grabber. Kodi has internal grabber. if you
      comment out the external grabber in your config - are your Hues
responding?
   3. Higher frequency / more lights will crash your set up at max. But it
   will run for a while. So make the hues light up first.

Thanks

**Comment by Portista1988** [comment]
Hello

1. i have the newer model. The link that you send i got error :(

2.1. See image this is what i get

![Test](https://user-images.githubusercontent.com/54452406/63647308-4c540880-c71f-11e9-967c-d9f5f350e192.jpg)

2.2 This is what i get when i try to screenhot

![Test Videograbber](https://user-images.githubusercontent.com/54452406/63647307-4c540880-c71f-11e9-9e2e-8fe62f3dc485.jpg)

I can't see nothing in the file after it

2.3 any help with this?
I try to set up lightberry if you mean but with that no response
Me i have the Hue Play Lamps

Was the file any good??

Hope to hear from you soon

thanx

**Comment by apurbas** [comment]
Issues related to Lightberry - you must speak to Tomek or Jacek @
Lightberry. You have paid for that service.

I see your problem. Your set up has Lightberry + Hue. I want you to
understand this & try accordingly:

   1. Controlling Lightberry & Hue requires Proto/Json forward. This means
   your Hyperion (..with Lightberry & external grabber) must forward the color
   codes to an external system (Philips Hue in your case) to be used by it.
   You will need another config file ( for Philips Hue) on a 2nd Raspberry Pi.
   2. I see Hyperion is installed properly. But Json port couldn't bind.
   This may be due to the port being used. Try this trick. On your config file
   - change the proto (say 19446) & Json ( say 19447) file to something
   different.
   3. Get your Lightberry to work flawlessly first. And then the Hue
   forwarder. Lightberry has a Kodi config plugin. Have you used that?
   4. On grabber - I'm not sure if your grabber can be seen on Pi. Try
   putting it on a powered USB hub connected to Raspberry Pi. Or the best, if
   it is provided by Lightberry team, get in touch with them for a resolution.

Cheers

**Comment by Portista1988** [comment]
The point is i have NO lightberry.
When i installed Openelec on mine raspberry i got the addon Lightberry
I tought was something i need to have so i tried to install that.
The only thing i have is the Hue Lamps

**Comment by apurbas** [comment]
Ok, I get you now.

Go ahead with creating a config file using Hyperion Config creator. Your
Hue bridge must be discoverable everytime. So give it a static Id to begin
with.

Let's make it work for Kodi to begin with. You can ignore/comment out the
external grabber codes etc in your Hyperion config file. This should be
easy.

Once we succeed here, will attempt to add external grabber.

If you run across proto/json error, try killing Hyperion from command line
& wait for a while. Or otherwise, try changing the ports in your Hyperion
config.

**Comment by apurbas** [comment]
I'm almost certain you have created wrong config. Please use Hyperion
config creator. The light types must be Philips Hue.
UART etc is not applicable for Hue.

**Comment by Portista1988** [comment]
Yeah i finally got the Hyperion config Creator. 
Now i have to adjust all

**Comment by apurbas** [comment]
Cheers mate!

**Comment by Portista1988** [comment]
I did everything but i see no results :(

I tried again to make a screenshot of the grabber and see now this:

![image](https://user-images.githubusercontent.com/54452406/63648495-ce98f880-c730-11e9-8798-6d038dafb73d.png)

Is already something more but i still have no idea

**Comment by Paulchen-Panther** [comment]
Comment the v4l2 session in you config out.

**Comment by Portista1988** [comment]
?

**Comment by apurbas** [comment]
Your config has external grabber code. Comment that out.

Make it work for Kodi first. ..

**Comment by Paulchen-Panther** [comment]
Lines begining with // are comments

**Comment by tpmodding** [comment]
please use the forum for support...
https://hyperion-project.org/
!

**Comment by Portista1988** [comment]
Something like this?

Without grabber??

[hyperion.config.zip](https://github.com/hyperion-project/hyperion/files/3538029/hyperion.config.zip)

**Comment by Paulchen-Panther** [comment]
yes

**Comment by Portista1988** [comment]
then i have still no succes
I put it on mine raspberry with the program WINSCP
I reboot after it and still mine lights are not changing at al

i still have the 3 lights on the way how i let it

**Comment by apurbas** [comment]
Config looks structurally OK.

Comment Out line 175 to 193 (external grabber section).  And run this
config file through command line as described earlier. Please post your
outcome then.

Is your Pi connected to TV via HDMI? with this setup - the Hue lights must
change with Kodi screen contents

All the best.

**Comment by Portista1988** [comment]
Sorry what do you mean?
Should i put the info again for the external?
Or now without the external grabber test the screenshot.

Sorry for a these questions i just want to fix it

**Comment by Portista1988** [comment]
by the way.

raspberry is connect to hdmi and aslo the grabber is still on it
so mine background should change to blue when the berry goes on

**Comment by apurbas** [comment]
No, let's fix it for Kodi/XBMC first.
Kodi has internal grabber. It should render your Hues.

**Comment by apurbas** [comment]
Absolutely my friend. Btw remove the USB grabber.

**Comment by Portista1988** [comment]
Oke i removed the Grabber now what?
I already placed the new file on the Pi and reboot it
I try the command ./autostart.sh but it will not accept it

**Comment by apurbas** [comment]
Ok, you need to learn the commands a bit for openelec. Run the following 2
Do you see rainbow swirl? Also what's the output you see on your Putty
window?

killall hyperiond

/storage/hyperion/bin/hyperiond.sh /storage/.config/hyperion.config.js

**Comment by Portista1988** [comment]
Dear sorry for mine late response.
Just got back

this is mine result after this 2 commands

![image](https://user-images.githubusercontent.com/54452406/63653548-826ca900-c76e-11e9-8aa8-7dc423a94f1d.png)

no i don't see any rainbow.
it is still the same 3 colors i adjust it when i turn it on
do i have to turn the light of first before starting the pi or should the be on??

**Comment by Paulchen-Panther** [comment]
hyperion.config.json **NOT** hyperion.config.js

**Comment by Portista1988** [comment]
Oke this is what i get
![image](https://user-images.githubusercontent.com/54452406/63653638-c4e2b580-c76f-11e9-9b5e-ecad2160c63d.png)

**Comment by Paulchen-Panther** [comment]
@Portista1988  
I will lock this thread because **THIS PLACE IS NOT THE FORUM.**
As @tpmodding  [already mentioned](https://github.com/hyperion-project/hyperion/issues/556#issuecomment-524620264). Please open under [https://hyperion-project.org/](https://hyperion-project.org/a) a new thread. There we can talk as long as we want.
THANK YOU.
